### PR TITLE
[TECHDEBT] Create Utility Configuration File (issue-#228)

### LINKED ISSUE #228: [TECHDEBT] Create Utility Configuration File
## Objective

Add an encapsulated config file to the utility module

## Origin Document

The Utility Module currently has **hard coded** configurations that should be dynamic based on the `config.json` file.

*See pattern of other module's config*

## Goals

- Allow dynamic configuration of the Utility Module

## Deliverable

- Code complete implementation
- Updated Changelog.md

## General issue deliverables

- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

**Creator**: @andrewnguyen22

### PR DESCRIPTION
**Pull request by andrewnguyen22**
## Description

The Utility Module currently has **hard coded** configurations that should be dynamic based on the `config.json` file.

*See pattern of other module's config*

- [x] Add an encapsulated config file to the utility module

## Issue

Fixes #228

## Type of change

Please mark the options that are relevant.

- [x] New feature, functionality or library
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other: <REPLACE_ME>

## List of changes

- Created UtilityConfig
- Added `Max_Mempool_Transaction_Bytes` and `Max_Mempool_Transactions` to the utility 
  config to allow dynamic configuration of the mempool
- Matched configuration unmarshalling pattern of other modules
- Added V0 mempool default configurations
- Regenerated build files with new mempool config

## Testing
- [x] `make test_all`
- [ x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my changes using the available tooling
- [ ] If applicable, I have made corresponding changes to related local or global README
- [ ] If applicable, I have added new diagrams using [mermaid.js](https://mermaid-js.github.io)
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: utility/doc/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.0.4] - 2022-09-23
+- Created UtilityConfig
+- Added `Max_Mempool_Transaction_Bytes` and `Max_Mempool_Transactions` to the utility
```
Comment (Olshansk) [inline utility/doc/CHANGELOG.md:12]: Snake case should use lowercase characters

File: utility/module.go [hunk 2]
```diff
@@ -21,14 +23,31 @@ const (
...
+	}
+	config := (c).(*types.UtilityConfig)
 	return &UtilityModule{
-		// TODO: Add `maxTransactionBytes` and `maxTransactions` to cfg.Utility
-		Mempool: types.NewMempool(1000, 1000),
+		Mempool: types.NewMempool(config.Max_Mempool_Transaction_Bytes, config.Max_Mempool_Transactions),
 	}, nil
 }
 
 func (u *UtilityModule) InitConfig(pathToConfigJSON string) (config modules.IConfig, err error) {
```
Comment (Olshansk) [inline utility/module.go:37]: I'm thinking we should aim to review & merge in #235 first and rebase this on top of the new pattern.

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: utility/module.go (hunk 2 above)
Comment (andrewnguyen22) [reply utility/module.go:37]: Why is that prioritized over this small one?

File: utility/doc/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.0.4] - 2022-09-23
+- Created UtilityConfig
```
Comment (Olshansk) [inline utility/doc/CHANGELOG.md:11]:
```suggestion
- Created `UtilityConfig`
```

File: utility/types/proto/util_config.proto [hunk 4]
```diff
@@ -4,5 +4,6 @@ package utility;
 option go_package = "github.com/pokt-network/pocket/utility/types";
 
 message UtilityConfig {
-
+  uint64 max_mempool_transaction_Bytes = 1;
```
Comment (Olshansk) [inline utility/types/proto/util_config.proto:7]:
```suggestion
  uint64 max_mempool_transaction_bytes = 1;
```

File: utility/module.go (hunk 2 above)
Comment (Olshansk) [reply utility/module.go:37]: Alright, let's prioritize this one first.

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Review by Olshansk — CHANGES_REQUESTED** [review]
Changes LGTM but there's a breaking change

Screen Shot 2022-09-29 at 2 52 55 PM

**Review by Olshansk — APPROVED** [review]
Can you fix the tests in `actor_test.go` before merging this in? Otherwise, LGTM.

Screen Shot 2022-10-03 at 3 04 58 PM

**Pull request merged**